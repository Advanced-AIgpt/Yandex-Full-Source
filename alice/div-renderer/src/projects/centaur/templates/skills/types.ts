import type { ISkillCardText } from './skillTypes/TextSkill/types';
import type { ISkillCardItemsList } from './skillTypes/ItemsListSkill/types';
import type { ISkillCardBigImage } from './skillTypes/BigImageSkill/types';
import { ISkillCardGallery } from './skillTypes/GallerySkill/types';

interface ISkillSuggest {
    text: string,
    url: string,
    payload: null | string,
}

export interface ISkillCardButton {
    text: string,
    url: string,
    payload: string | null,
}

export type ISkillCardType = ISkillCardText | ISkillCardItemsList | ISkillCardBigImage | ISkillCardGallery;

export type CardType = ISkillCardType['type'];

export interface Types<T = ISkillCardType> {
    skillInfo: {
        name: string,
        logo: string,
        dialogId: string,
    },
    request: {
        text: string,
    },
    response: {
        card: T,
        buttons: ISkillCardButton[],
        suggests: ISkillSuggest[],
    },
}
