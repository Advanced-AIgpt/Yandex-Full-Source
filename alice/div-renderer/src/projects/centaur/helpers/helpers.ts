/* eslint-disable camelcase */
import { sortedIndexBy } from 'lodash';
import { format, isSameDay } from 'date-fns';
import { ru } from 'date-fns/locale';
import { Div, IDivData, TextBlock } from 'divcard2';
import { IRequestState } from '../../../common/types/common';
import { getTemplates, templatesRegistry } from '../../../registries/common';
import { centaurTemplatesClass } from '../index';
import { Palette } from './palette';

export { default as pluralize } from '@yandex-int/i18n/lib/plural/ru';
type TextProps = Pick<ConstructorParameters<typeof TextBlock>[0], 'text_color'|'font_size'|'font_weight'|'line_height'>;

export const textBreakpoint =
    (...breakpoints: [number, TextProps][]) =>
        (text: string) => {
            const idx = Math.min(sortedIndexBy(breakpoints, [text.length, {}], 0), breakpoints.length - 1);
            return { text, ...breakpoints[idx][1] };
        };

export const formatDate = (d: Date | string | number, f: string) => format(new Date(d), f, { locale: ru });

export const textCommand = (c: string) => `dialog://text_command?query=${encodeURI(c)}`;

export const updatedAt = (d: Date) => {
    const now = new Date();
    return `Обновлен ${isSameDay(now, d) ? 'сегодня' : formatDate(d, 'dd.MM')}`;
};

export const insertBetween = <I extends unknown, S extends unknown>(itemList: I[], separator: S) => {
    return itemList.reduce<(I | S)[]>((result, item, index) => {
        if (index !== 0) {
            result.push(separator);
        }

        result.push(item);

        return result;
    }, []);
};

export interface ITemplateCard {
    templates: {
        [name: string]: Div;
    };
    palette?: Palette;
    card: IDivData;
}

// Дива верхнего уровня. Замена для new `TemplateCard()`
export const TopLevelCard = (card: Omit<IDivData, 'variable_triggers' | 'variables'>, requestState: IRequestState, palette?: Palette) => {
    const templates = Object.assign(
        getTemplates(requestState.localTemplates, templatesRegistry),
        centaurTemplatesClass.getUsedTemplates(requestState),
        requestState.res.localTemplates.getAll(),
    );

    return {
        templates,
        palette,
        card: {
            ...card,
            variable_triggers: requestState.variableTriggers.getAll(),
            variables: [...requestState.variables],
        } as IDivData,
    };
};
