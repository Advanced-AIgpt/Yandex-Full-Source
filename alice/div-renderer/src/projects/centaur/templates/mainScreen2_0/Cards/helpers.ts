export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
