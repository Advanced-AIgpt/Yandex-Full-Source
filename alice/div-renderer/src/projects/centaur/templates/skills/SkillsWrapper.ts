import {
    Div,
} from 'divcard2';
import FullScreen from '../../components/FullScreen';
import TitleLine from '../../components/TitleLine/TitleLine';
import { SuggestsBlockNormal } from '../suggests';
import {
    gradientToBlackBottom,
    gradientToBlackTop,
    offsetFromEdgeOfScreen,
} from '../../style/constants';
import { Types } from './types';
import { defaultActionsCloseLayer } from '../../components/CloseButton';
import { directivesAction } from '../../../../common/actions';
import { createSemanticFrameAction } from '../../../../common/actions/server';

export const DeviceName = 'SmartSpeaker';

interface Props {
    data: Readonly<Types>;
    children: Readonly<Div[]>;
}

export default function SkillsWrapper({ data, children }: Props) {
    const { skillInfo, response } = data;

    const skillName = skillInfo.name;
    const skillLogo = skillInfo.logo;
    const suggests = response.suggests;

    return FullScreen({
        children: [
            ...children,
            TitleLine({
                title: skillName,
                icon: skillLogo,
                needClose: true,
                options: { background: gradientToBlackTop },
                closeButtonOptions: {
                    options: {
                        actions: [
                            {
                                log_id: 'action_close_skill',
                                url: directivesAction(
                                    createSemanticFrameAction(
                                        {
                                            external_skill_force_deactivate_semantic_frame: {
                                                dialog_id: {
                                                    string_value: skillInfo.dialogId,
                                                },
                                                silent_response: {
                                                    bool_value: true,
                                                },
                                            },
                                        },
                                        'Dialogovo',
                                        DeviceName,
                                        'alice.external_skill_force_deactivate',
                                    ),
                                ),
                            },
                            ...defaultActionsCloseLayer,
                        ],
                    },
                },
            }),
            SuggestsBlockNormal(
                suggests.map(el => ({ Title: el.text, ActionId: el.url, transparent: false })),
                {
                    alignment_vertical: 'bottom',
                    alignment_horizontal: 'left',
                    paddings: {
                        left: offsetFromEdgeOfScreen,
                        right: offsetFromEdgeOfScreen,
                        bottom: offsetFromEdgeOfScreen,
                    },
                    background: gradientToBlackBottom,
                },
            ),
        ],
        options: {
            orientation: 'overlap',
        },
    });
}
