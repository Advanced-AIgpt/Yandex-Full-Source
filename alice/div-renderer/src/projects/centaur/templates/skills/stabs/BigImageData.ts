import { NAlice } from '../../../../../protos';
type ITDialogovoSkillCardData = NAlice.NData.ITDialogovoSkillCardData;

export const params: ITDialogovoSkillCardData = {
    SkillInfo: {
        Name: 'Мистический Петербург',
        Logo: 'https://avatars.mds.yandex.net/get-dialogs/998463/ce24f9c3c26fd8375668/mobile-logo-x4',
    },
    SkillRequest: {
        Text: 'историческая мозаика',
    },
    SkillResponse: {
        BigImageResponse: {
            ImageItem: {
                ImageUrl: 'https://avatars.mds.yandex.net/get-dialogs-skill-card/937455/eed017c000a2aab9ed6f/one-x4',
                Title: 'Историческая мозаика (7+)',
                Description: 'Квест в формате экскурсионной прогулки проходит по центру Петербурга, от Екатерининского сада до Невского проспекта. Вы сможете посетить самые интересные места, а мои головоломки позволят погрузиться в историю города. Подойдет как для туристов, так и для жителей города.',
                Button: null,
            },
        },
        buttons: [
            {
                Text: 'Подробное описание',
                Url: 'dialog://text_command?query=Подробное описание',
                Payload: null,
            },
            {
                Text: 'Начать квест',
                Url: 'dialog://text_command?query=Начать квест',
                Payload: null,
            },
        ],
        suggests: [
            {
                Text: 'Меню',
                Url: 'dialog://text_command?query=Меню',
                Payload: null,
            },
            {
                Text: 'Все квесты',
                Url: 'dialog://text_command?query=Все квесты',
                Payload: null,
            },
            {
                Text: 'Мои квесты',
                Url: 'dialog://text_command?query=Мои квесты',
                Payload: null,
            },
            {
                Text: 'Авторизация',
                Url: 'dialog://text_command?query=Авторизация',
                Payload: null,
            },
        ],
    },
};
