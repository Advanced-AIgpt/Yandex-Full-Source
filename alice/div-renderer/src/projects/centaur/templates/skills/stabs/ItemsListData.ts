import { NAlice } from '../../../../../protos';
type ITDialogovoSkillCardData = NAlice.NData.ITDialogovoSkillCardData;

export const params: ITDialogovoSkillCardData = {
    SkillInfo: {
        Name: 'Мистический Петербург',
        Logo: 'https://avatars.mds.yandex.net/get-dialogs/998463/ce24f9c3c26fd8375668/orig',
    },
    SkillRequest: {
        Text: 'квесты',
    },
    SkillResponse: {
        ItemsListResponse: {
            ItemsLisetHeader: {
                Text: null,
            },
            ImageItems: [
                {
                    ImageUrl: 'https://avatars.mds.yandex.net/get-dialogs-skill-card/213044/923f22cc2a912ddd68e4/orig',
                    Title: 'Историческая мозаика',
                    Description: 'беплатно, 1-10 чел., от 7 лет',
                    Button: {
                        Text: 'Историческая мозаика',
                        Url: 'dialog://text_command?query=Историческая мозаика',
                        Payload: null,
                    },
                },
                {
                    ImageUrl: 'https://avatars.mds.yandex.net/get-dialogs-skill-card/213044/923f22cc2a912ddd68e4/orig',
                    Title: 'По следам Пиковой Дамы',
                    Description: 'Цена: 3000 руб., 1-5 чел., от 7 лет',
                    Button: {
                        Text: 'По следам Пиковой Дамы',
                        Url: 'dialog://text_command?query=По следам Пиковой Дамы',
                        Payload: null,
                    },
                },
                {
                    ImageUrl: 'https://avatars.mds.yandex.net/get-dialogs-skill-card/213044/923f22cc2a912ddd68e4/orig',
                    Title: 'Безумие Алисы',
                    Description: 'Цена: 4000 руб., 2-7 чел., от 16 лет',
                    Button: {
                        Text: 'Безумие Алисы',
                        Url: 'dialog://text_command?query=Безумие Алисы',
                        Payload: null,
                    },
                },
                {
                    ImageUrl: 'https://avatars.mds.yandex.net/get-dialogs-skill-card/213044/923f22cc2a912ddd68e4/orig',
                    Title: 'Безумие Алисы',
                    Description: 'Цена: 4000 руб., 2-7 чел., от 16 лет',
                    Button: {
                        Text: 'Безумие Алисы',
                        Url: 'dialog://text_command?query=Безумие Алисы',
                        Payload: null,
                    },
                },
                {
                    ImageUrl: 'https://avatars.mds.yandex.net/get-dialogs-skill-card/213044/923f22cc2a912ddd68e4/orig',
                    Title: 'Безумие Алисы',
                    Description: 'Цена: 4000 руб., 2-7 чел., от 16 лет',
                    Button: {
                        Text: 'Безумие Алисы',
                        Url: 'dialog://text_command?query=Безумие Алисы',
                        Payload: null,
                    },
                },
                {
                    ImageUrl: 'https://avatars.mds.yandex.net/get-dialogs-skill-card/213044/923f22cc2a912ddd68e4/orig',
                    Title: 'Безумие Алисы',
                    Description: 'Цена: 4000 руб., 2-7 чел., от 16 лет',
                    Button: {
                        Text: 'Безумие Алисы',
                        Url: 'dialog://text_command?query=Безумие Алисы',
                        Payload: null,
                    },
                },
            ],
            ItemsLisetFooter: {
                Text: null,
                Button: null,
            },
        },
        buttons: [],
        suggests: [
            {
                Text: 'Меню',
                Url: 'dialog://text_command?query=Меню',
                Payload: null,
            },
            {
                Text: 'Все квесты',
                Url: 'dialog://text_command?query=Все квесты',
                Payload: null,
            },
            {
                Text: 'Мои квесты',
                Url: 'dialog://text_command?query=Мои квесты',
                Payload: null,
            },
            {
                Text: 'Авторизация',
                Url: 'dialog://text_command?query=Авторизация',
                Payload: null,
            },
        ],
    },
};
