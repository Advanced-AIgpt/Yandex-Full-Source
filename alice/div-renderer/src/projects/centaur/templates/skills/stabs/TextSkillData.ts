import { NAlice } from '../../../../../protos';
type ITDialogovoSkillCardData = NAlice.NData.ITDialogovoSkillCardData;

export const params: ITDialogovoSkillCardData = {
    SkillInfo: {
        Name: 'Мистический Петербург',
        Logo: 'https://avatars.mds.yandex.net/get-dialogs/998463/ce24f9c3c26fd8375668/orig',
    },
    SkillRequest: {
        Text: 'начать квест',
    },
    SkillResponse: {
        TextResponse: {
            Text: 'Для начала квеста, уточните пожалуйста сколько',
        },
        buttons: [],
        suggests: [
            {
                Text: 'Меню',
                Url: 'dialog://text_command?query=Меню',
                Payload: null,
            },
            {
                Text: 'Все квесты',
                Url: 'dialog://text_command?query=Все квесты',
                Payload: null,
            },
            {
                Text: 'Мои квесты',
                Url: 'dialog://text_command?query=Мои квесты',
                Payload: null,
            },
            {
                Text: 'Авторизация',
                Url: 'dialog://text_command?query=Авторизация',
                Payload: null,
            },
        ],
    },
};
