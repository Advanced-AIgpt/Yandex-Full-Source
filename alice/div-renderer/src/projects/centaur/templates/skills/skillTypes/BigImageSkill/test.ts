import skills from '../../index';
import { params } from '../../stabs/BigImageData';
import { AnonymizeDataForSnapshot } from '../../../../../../common/helpers/dev';

jest.mock('../../../../../../common/logger');

describe('BigImageSkill', () => {
    it('should equal snapshot', () => {
        expect(AnonymizeDataForSnapshot(skills(params))).toMatchSnapshot();
    });
});
