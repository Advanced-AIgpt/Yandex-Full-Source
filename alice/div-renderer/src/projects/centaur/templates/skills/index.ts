import { Div, TemplateCard, Templates } from 'divcard2';
import { compact } from 'lodash';
import { suggestButtonHeight } from '../suggests';
import { offsetFromEdgeOfScreen } from '../../style/constants';
import SkillsWrapper from './SkillsWrapper';
import { CardType, ISkillCardType, Types } from './types';
import TextSkill from './skillTypes/TextSkill/TextSkill';
import ItemsListSkill from './skillTypes/ItemsListSkill/ItemsListSkill';
import BigImageSkill from './skillTypes/BigImageSkill/BigImageSkill';
import { NAlice } from '../../../../protos';
import { getTextData } from './skillTypes/TextSkill/getData';
import { getItemListData } from './skillTypes/ItemsListSkill/getData';
import { getBigImageData } from './skillTypes/BigImageSkill/getData';
import GallerySkill from './skillTypes/GallerySkill/GallerySkill';
import { getGalleryData } from './skillTypes/GallerySkill/getData';
type ITDialogovoSkillCardData = NAlice.NData.ITDialogovoSkillCardData;

// в будущем должно приходить в запросе
const clientHeight = 880;
const suggestsBlockHeight = offsetFromEdgeOfScreen * 2 + suggestButtonHeight;

const componentContentMap: {[name in CardType]: (options: {
    data: Types,
    suggestsBlockHeight: number,
    clientHeight: number,
}) => Div} = {
    Text: TextSkill,
    BigImage: BigImageSkill,
    ItemsList: ItemsListSkill,
    Gallery: GallerySkill,
};

function cardAdapter(card: NAlice.NData.TDialogovoSkillCardData.ITSkillResponse): ISkillCardType | null {
    return getTextData(card) ??
        getItemListData(card) ??
        getBigImageData(card) ??
        getGalleryData(card);
}

export function skillsDataAdapter(data: ITDialogovoSkillCardData): Types {
    let cardInfo = data.SkillResponse && cardAdapter(data.SkillResponse);

    if (!cardInfo) {
        cardInfo = {
            type: 'Text',
            text: 'Ошибка. Не удается отобразить содержимое.',
        };
    }

    return {
        skillInfo: {
            name: data.SkillInfo?.Name || ' ',
            logo: data.SkillInfo?.Logo || '',
            dialogId: data.SkillInfo?.SkillId || '',
        },
        response: {
            card: cardInfo,
            buttons: compact(data.SkillResponse?.buttons?.map(
                button => (
                    button.Url && button.Text ?
                        {
                            url: button.Url,
                            text: button.Text,
                            payload: button.Payload || null,
                        } : null
                ),
            )) ||
                [],
            suggests: compact(data.SkillResponse?.suggests?.map(
                suggest => (
                    suggest.Url && suggest.Text ?
                        {
                            url: suggest.Url,
                            text: suggest.Text,
                            payload: suggest.Payload || null,
                        } : null
                ),
            )) ||
                [],
        },
        request: {
            text: data.SkillRequest?.Text || 'Навык',
        },
    };
}

export default function skills(data: ITDialogovoSkillCardData) {
    const params = skillsDataAdapter(data);
    const content = componentContentMap[params.response.card.type]({
        data: params,
        clientHeight,
        suggestsBlockHeight,
    });

    return new TemplateCard(new Templates({}), {
        log_id: 'skill_card',
        states: [
            {
                state_id: 0,
                div: SkillsWrapper({
                    data: params,
                    children: [content],
                }),
            },
        ],
    });
}
