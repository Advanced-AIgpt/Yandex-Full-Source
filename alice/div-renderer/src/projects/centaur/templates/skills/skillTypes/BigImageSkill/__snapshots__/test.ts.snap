// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigImageSkill should equal snapshot 1`] = `
Object {
  "card": Object {
    "log_id": "skill_card",
    "states": Array [
      Object {
        "div": Object {
          "background": Array [
            Object {
              "color": "#121316",
              "type": "solid",
            },
          ],
          "height": Object {
            "type": "match_parent",
          },
          "items": Array [
            Object {
              "height": Object {
                "type": "fixed",
                "value": 880,
              },
              "items": Array [
                Object {
                  "border": Object {
                    "corner_radius": 28,
                  },
                  "image_url": "https://avatars.mds.yandex.net/get-dialogs-skill-card/937455/eed017c000a2aab9ed6f/one-x4",
                  "margins": Object {
                    "bottom": 32,
                    "left": 48,
                    "right": 48,
                  },
                  "type": "image",
                  "width": Object {
                    "type": "fixed",
                    "value": 554,
                  },
                },
                Object {
                  "font_size": 44,
                  "font_weight": "regular",
                  "line_height": 52,
                  "margins": Object {
                    "left": 48,
                    "right": 48,
                  },
                  "text": "Историческая мозаика (7+)

Квест в формате экскурсионной прогулки проходит по центру Петербурга, от Екатерининского сада до Невского проспекта. Вы сможете посетить самые интересные места, а мои головоломки позволят погрузиться в историю города. Подойдет как для туристов, так и для жителей города.",
                  "text_color": "#e6ffffff",
                  "type": "text",
                },
                Object {
                  "item_spacing": 24,
                  "items": Array [
                    Object {
                      "background": Array [
                        Object {
                          "color": "#fff",
                          "type": "solid",
                        },
                      ],
                      "border": Object {
                        "corner_radius": 28,
                      },
                      "height": Object {
                        "type": "wrap_content",
                      },
                      "items": Array [
                        Object {
                          "action": Object {
                            "log_id": "button_list_button",
                            "url": "dialog://text_command?query=Подробное описание",
                          },
                          "font_size": 44,
                          "font_weight": "regular",
                          "line_height": 52,
                          "text": "Подробное описание",
                          "text_color": "#121316",
                          "type": "text",
                        },
                      ],
                      "orientation": "horizontal",
                      "paddings": Object {
                        "bottom": 29,
                        "left": 40,
                        "right": 40,
                        "top": 29,
                      },
                      "type": "container",
                      "width": Object {
                        "type": "wrap_content",
                      },
                    },
                    Object {
                      "background": Array [
                        Object {
                          "color": "#fff",
                          "type": "solid",
                        },
                      ],
                      "border": Object {
                        "corner_radius": 28,
                      },
                      "height": Object {
                        "type": "wrap_content",
                      },
                      "items": Array [
                        Object {
                          "action": Object {
                            "log_id": "button_list_button",
                            "url": "dialog://text_command?query=Начать квест",
                          },
                          "font_size": 44,
                          "font_weight": "regular",
                          "line_height": 52,
                          "text": "Начать квест",
                          "text_color": "#121316",
                          "type": "text",
                        },
                      ],
                      "orientation": "horizontal",
                      "paddings": Object {
                        "bottom": 29,
                        "left": 40,
                        "right": 40,
                        "top": 29,
                      },
                      "type": "container",
                      "width": Object {
                        "type": "wrap_content",
                      },
                    },
                  ],
                  "margins": Object {
                    "top": 48,
                  },
                  "orientation": "horizontal",
                  "paddings": Object {
                    "left": 48,
                    "right": 48,
                  },
                  "type": "gallery",
                },
              ],
              "orientation": "vertical",
              "paddings": Object {
                "bottom": 162,
                "top": 184,
              },
              "type": "gallery",
            },
            Object {
              "alignment_horizontal": "left",
              "alignment_vertical": "top",
              "background": Array [
                Object {
                  "angle": 270,
                  "colors": Array [
                    "#e6000000",
                    "#a0050607",
                    "#00121316",
                  ],
                  "type": "gradient",
                },
              ],
              "content_alignment_vertical": "center",
              "items": Array [
                Object {
                  "border": Object {
                    "corner_radius": 44,
                  },
                  "height": Object {
                    "type": "fixed",
                    "value": 88,
                  },
                  "image_url": "https://avatars.mds.yandex.net/get-dialogs/998463/ce24f9c3c26fd8375668/mobile-logo-x4",
                  "margins": Object {
                    "right": 32,
                  },
                  "type": "image",
                  "width": Object {
                    "type": "fixed",
                    "value": 88,
                  },
                },
                Object {
                  "font_size": 40,
                  "font_weight": "medium",
                  "line_height": 56,
                  "text": "Мистический Петербург",
                  "text_color": "#88898A",
                  "type": "text",
                },
                Object {
                  "actions": Array [
                    Object {
                      "log_id": "action_close_skill",
                      "url": "dialog://?directives=%5B%7B%22type%22%3A%22server_action%22%2C%22name%22%3A%22%40%40mm_semantic_frame%22%2C%22payload%22%3A%7B%22typed_semantic_frame%22%3A%7B%22external_skill_force_deactivate_semantic_frame%22%3A%7B%22dialog_id%22%3A%7B%22string_value%22%3A%22%22%7D%2C%22silent_response%22%3A%7B%22bool_value%22%3Atrue%7D%7D%7D%2C%22analytics%22%3A%7B%22product_scenario%22%3A%22Dialogovo%22%2C%22origin%22%3A%22alice.external_skill_force_deactivate%22%2C%22purpose%22%3A%22SmartSpeaker%22%7D%2C%22params%22%3A%7B%22disable_output_speech%22%3Atrue%2C%22disable_should_listen%22%3Atrue%7D%7D%7D%5D",
                    },
                    Object {
                      "log_id": "close_fullscreen",
                      "url": "centaur://local_command?local_commands=[%7B%22command%22%3A%22close_layer%22%2C%22layer%22%3A%22DIALOG%22%2C%22stop_conversation%22%3Atrue%7D]",
                    },
                    Object {
                      "log_id": "close_fullscreen",
                      "url": "centaur://local_command?local_commands=[%7B%22command%22%3A%22close_layer%22%2C%22layer%22%3A%22DIALOG%22%2C%22stop_conversation%22%3Atrue%7D]",
                    },
                  ],
                  "alignment_horizontal": "right",
                  "background": Array [
                    Object {
                      "color": "#1affffff",
                      "type": "solid",
                    },
                  ],
                  "border": Object {
                    "corner_radius": 20,
                  },
                  "height": Object {
                    "type": "fixed",
                    "value": 88,
                  },
                  "image_url": "https://yastatic.net/s3/dialogs/smart_displays/icons/close_wt_border.png",
                  "paddings": Object {
                    "bottom": 22,
                    "left": 22,
                    "right": 22,
                    "top": 22,
                  },
                  "preload_required": 1,
                  "scale": "fill",
                  "type": "image",
                  "width": Object {
                    "type": "fixed",
                    "value": 88,
                  },
                },
              ],
              "orientation": "horizontal",
              "paddings": Object {
                "bottom": 48,
                "left": 48,
                "right": 48,
                "top": 48,
              },
              "type": "container",
            },
            Object {
              "alignment_horizontal": "left",
              "alignment_vertical": "bottom",
              "background": Array [
                Object {
                  "angle": 90,
                  "colors": Array [
                    "#e6000000",
                    "#a0050607",
                    "#00121316",
                  ],
                  "type": "gradient",
                },
              ],
              "item_spacing": 24,
              "items": Array [
                Object {
                  "actions": Array [
                    Object {
                      "log_id": "dialog://text_command?query=Меню",
                      "url": "dialog://text_command?query=%D0%9C%D0%B5%D0%BD%D1%8E",
                    },
                  ],
                  "background": Array [
                    Object {
                      "color": "#292A2D",
                      "type": "solid",
                    },
                  ],
                  "border": Object {
                    "corner_radius": 33,
                  },
                  "font_size": 36,
                  "font_weight": "medium",
                  "height": Object {
                    "type": "wrap_content",
                  },
                  "line_height": 46,
                  "paddings": Object {
                    "bottom": 13,
                    "left": 24,
                    "right": 24,
                    "top": 13,
                  },
                  "text": "Меню",
                  "text_color": "#fff",
                  "type": "text",
                  "width": Object {
                    "type": "wrap_content",
                  },
                },
                Object {
                  "actions": Array [
                    Object {
                      "log_id": "dialog://text_command?query=Все квесты",
                      "url": "dialog://text_command?query=%D0%92%D1%81%D0%B5%20%D0%BA%D0%B2%D0%B5%D1%81%D1%82%D1%8B",
                    },
                  ],
                  "background": Array [
                    Object {
                      "color": "#292A2D",
                      "type": "solid",
                    },
                  ],
                  "border": Object {
                    "corner_radius": 33,
                  },
                  "font_size": 36,
                  "font_weight": "medium",
                  "height": Object {
                    "type": "wrap_content",
                  },
                  "line_height": 46,
                  "paddings": Object {
                    "bottom": 13,
                    "left": 24,
                    "right": 24,
                    "top": 13,
                  },
                  "text": "Все квесты",
                  "text_color": "#fff",
                  "type": "text",
                  "width": Object {
                    "type": "wrap_content",
                  },
                },
                Object {
                  "actions": Array [
                    Object {
                      "log_id": "dialog://text_command?query=Мои квесты",
                      "url": "dialog://text_command?query=%D0%9C%D0%BE%D0%B8%20%D0%BA%D0%B2%D0%B5%D1%81%D1%82%D1%8B",
                    },
                  ],
                  "background": Array [
                    Object {
                      "color": "#292A2D",
                      "type": "solid",
                    },
                  ],
                  "border": Object {
                    "corner_radius": 33,
                  },
                  "font_size": 36,
                  "font_weight": "medium",
                  "height": Object {
                    "type": "wrap_content",
                  },
                  "line_height": 46,
                  "paddings": Object {
                    "bottom": 13,
                    "left": 24,
                    "right": 24,
                    "top": 13,
                  },
                  "text": "Мои квесты",
                  "text_color": "#fff",
                  "type": "text",
                  "width": Object {
                    "type": "wrap_content",
                  },
                },
                Object {
                  "actions": Array [
                    Object {
                      "log_id": "dialog://text_command?query=Авторизация",
                      "url": "dialog://text_command?query=%D0%90%D0%B2%D1%82%D0%BE%D1%80%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F",
                    },
                  ],
                  "background": Array [
                    Object {
                      "color": "#292A2D",
                      "type": "solid",
                    },
                  ],
                  "border": Object {
                    "corner_radius": 33,
                  },
                  "font_size": 36,
                  "font_weight": "medium",
                  "height": Object {
                    "type": "wrap_content",
                  },
                  "line_height": 46,
                  "paddings": Object {
                    "bottom": 13,
                    "left": 24,
                    "right": 24,
                    "top": 13,
                  },
                  "text": "Авторизация",
                  "text_color": "#fff",
                  "type": "text",
                  "width": Object {
                    "type": "wrap_content",
                  },
                },
              ],
              "paddings": Object {
                "bottom": 48,
                "left": 48,
                "right": 48,
              },
              "type": "gallery",
            },
          ],
          "orientation": "overlap",
          "type": "container",
          "width": Object {
            "type": "match_parent",
          },
        },
        "state_id": 0,
      },
    ],
  },
  "templates": Object {},
}
`;
